projects/packages/search/src/dashboard/components/record-meter/bar-chart.jsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import Chart from 'chart.js/auto';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+export class BarChart extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.canvasRef = React.createRef();
+	}
+
+	componentDidMount() {
+		this.myChart = new Chart( this.canvasRef.current, {
+			type: 'bar',
+			options: {
+				borderRadius: 100,
+				borderSkipped: 'middle',
+				indexAxis: 'y',
+				maintainAspectRatio: false,
+				aspectRatio: 1.3,
+				scales: {
+					x: {
+						stacked: true,
+						grid: {
+							display: false,
+							drawBorder: false,
+						},
+						ticks: {
+							display: false,
+						},
+					},
+					y: {
+						stacked: true,
+						grid: {
+							display: false,
+							drawBorder: false,
+						},
+						ticks: {
+							display: false,
+						},
+					},
+				},
+
+				plugins: {
+					title: {
+						display: false,
+					},
+					subtitle: {
+						display: false,
+					},
+					legend: {
+						display: false,
+
+						labels: {
+							filter: function ( legendItem ) {
+								return ! legendItem.text.includes( 'Remaining' );
+							},
+						},
+					},
+				},
+			},
+
+			data: {
+				labels: [ '' ], // this empty label must remain, if removed the entire chart breaks
+				datasets: this.props.data.map( d => d.data ),
+			},
+		} );
+
+		if ( this.props.isValid === true ) {
+			// TODO: move setState out of componentDidMount
+			// eslint-disable-next-line react/no-did-mount-set-state
+			this.setState( { legendItems: this.myChart.legend.legendItems } );
+		}
+	}
+
+	render() {
+		if ( this.props.isValid === false ) {
+			return null;
+		}
+		return (
+			<div className="jp-search-bar-chart">
+				<div className="jp-search-bar-chart__container">
+					<canvas ref={ this.canvasRef }>
+						<p>
+							{ __(
+								'Text alternative for this canvas graphic is in the data table below',
```

Comment:
Maybe "for this chart" rather than "for this canvas graphic" which feels a bit too technical?